pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -185,25 +193,41 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
 
 	managementCluster, err := utils.GetManagementCluster(ctx, r.Client)
 	if err != nil {
-		log.Error(err, "Error reading ManagementCluster")
-		r.status.SetDegraded("Error reading ManagementCluster", err.Error())
+		status.SetDegraded(r.status, operatorv1.ResourceReadError, "Error reading ManagementCluster", err, reqLogger)
 		return reconcile.Result{}, err
 	}
 
 	// Fetch the managementClusterConnection.
 	managementClusterConnection, err := utils.GetManagementClusterConnection(ctx, r.Client)
 	if err != nil {
-		r.status.SetDegraded("Error querying ManagementClusterConnection", err.Error())
+		status.SetDegraded(r.status, operatorv1.ResourceReadError, "Error querying ManagementClusterConnection", err, reqLogger)
 		return result, err
 	} else if managementClusterConnection == nil {
 		r.status.OnCRNotFound()
 		return result, nil
 	}
+	// SetMetaData in the TigeraStatus like observedGenerations.
```

Comment:
I think the line `r.status.OnCRFound()` could be moved to the line above this line.